src/foundry/client-esm/applications/ux/text-editor.d.mts

```diff
@@ -1,9 +1,530 @@
+   */
+  protected static _onClickInlineRoll(event: MouseEvent): void;
+
+  /**
+   * Extract JSON data from a drag/drop event.
+   * @param event - The drag event which contains JSON data.
+   * @returns The extracted JSON data. The object will be empty if the DragEvent did not contain JSON-parseable data.
+   * @remarks `TokensLayer#_onDropActorData` returns a number - a notification ID - the  if the user lacks permissions
+   * @remarks `User#assignHotbarMacro` returns a promise to a user document
+   */
+  static getDragEventData(event: DragEvent): JSONValue;
+
+  /**
+   * Given a Drop event, returns a Content link if possible such as `"@Actor[ABC123]"`, else `null`
+   * @param eventData - The parsed object of data provided by the transfer event
+   * @param options   - Additional options to configure link creation.
+   */
+  static getContentLink(eventData: JSONValue, options?: TextEditor.GetContentLinkOptions): Promise<string | null>;
+
+  /**
+   * Upload an image to a document's asset path.
+   * @param uuid - The document's UUID.
+   * @param file - The image file to upload.
+   * @returns The path to the uploaded image.
+   * @internal
+   */
+  protected static _uploadImage(uuid: string, file: File): Promise<string | void>;
+
+  /**
+   * Retrieve the configured TextEditor implementation.
+   * @remarks TODO: Link up with CONFIG
+   */
+  static get implementation(): typeof TextEditor;
+
+  #private: true;
+}
+
+declare namespace TextEditor {
+  interface Options<Engine extends "tinymce" | "prosemirror"> {
+    /**
+     * Which rich text editor engine to use, "tinymce" or "prosemirror". TinyMCE
+     * is deprecated and will be removed in a later version.
+     * @defaultValue `"tinymce"`
+     */
+    engine?: Engine;
+
+    /**
+     * @defaultValue `false`
+     */
+    branding?: boolean;
+
+    /**
+     * @defaultValue `["/css/mce.css"]`
+     */
+    content_css?: string[];
+
+    /**
+     * @defaultValue `false`
+     */
+    menubar?: boolean;
+
+    /**
+     * @defaultValue `"lists image table hr code save link"`
+     */
+    plugins?: string;
+
+    /**
+     * @defaultValue `true`
+     */
+    save_enablewhendirty?: boolean;
+
+    /**
+     * @defaultValue `false`
+     */
+    statusbar?: boolean;
+
+    style_formats?: [
+      {
+        items?: [
+          {
+            /**
+             * @defaultValue `"section"`
+             */
+            block?: string;
+
+            /**
+             * @defaultValue `"secrect"`
+             */
+            classes?: string;
+
+            /**
+             * @defaultValue `"Secret"`
+             */
+            title?: string;
+
+            /**
+             * @defaultValue `true`
+             */
+            wrapper?: boolean;
+          },
+        ];
+
+        /**
+         * @defaultValue `"Custom"`
+         */
+        title?: string;
+      },
+    ];
+
+    /**
+     * @defaultValue `true`
+     */
+    style_formats_merge?: boolean;
```

Comment:
Noting here because it's convenient but applies to all properties. Does `null` and `undefined` apply here?